schema_based_on_diagram.sql

```diff
@@ -0,0 +1,50 @@
+CREATE TABLE patients (
+  id INT NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  name VARCHAR(100),
+  date_of_birth DATE
+);
+
+CREATE TABLE treatments (
+  id INT NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  type VARCHAR(100),
+  name VARCHAR(100)
+);
+
+CREATE TABLE medical_histories (
+  id INT NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  admitted_at TIMESTAMP,
+  patient_id INT REFERENCES patients(id),
+  status VARCHAR(100)
+);
+
+CREATE TABLE invoices (
+  id INT NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history_id INT REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items (
+  id INT NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT REFERENCES invoices(id),
+  treatment_id INT REFERENCES treatments(id)
+);
+
+CREATE TABLE treatment_history (
+  medical_history_id INT REFERENCES medical_histories(id),
+  treatment_id INT REFERENCES treatments(id),
+  PRIMARY KEY (medical_history_id, treatment_id),
+  CONSTRAINT fk_treatment_history
+    FOREIGN KEY(medical_history_id) REFERENCES medical_histories(id) ON DELETE CASCADE,
+    FOREIGN KEY(treatment_id) REFERENCES treatments(id) ON DELETE CASCADE
+);
+
+CREATE INDEX patient_id_asc ON patients(id ASC);
+CREATE INDEX treatments_id_asc ON treatments(id ASC);
+CREATE INDEX medical_history_id_asc ON medical_histories(id ASC);
+CREATE INDEX invoice_items_id_asc ON invoice_items(invoice_id ASC);
+CREATE INDEX treatment_history_id_asc ON treatment_history(treatment_id ASC);
```

Comment:
- [ Required ] An index is usually added `FOREIGN KEYS` not primary keys, so it would be best practice if you add an index on all `foreign keys`  as shown in the example below.

```SQL
CREATE INDEX ON medical_histories (patient_id);
CREATE INDEX ON invoices (medical_history_id);
CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON invoice_items (treatment_id);
CREATE INDEX ON treatment_history(medical_history_id);
CREATE INDEX ON treatment_history(treatment_id);
```

Reply:
Thanks for the suggestion. Working on it!👍